dao/src/main/scala/za/co/absa/enceladus/dao/rest/CrossHostApiCaller.scala

```diff
@@ -48,24 +49,34 @@ object CrossHostApiCaller {
       urlsRetryCount
     }) + 1
     val currentHostIndex = startWith.getOrElse(Random.nextInt(Math.max(apiBaseUrls.size, 1)))
-    new CrossHostApiCaller(apiBaseUrls.toVector, maxTryCount, currentHostIndex, optionallyRetryableExceptions)
+    new CrossHostApiCaller(
+      apiBaseUrls.toVector, maxTryCount, currentHostIndex, optionallyRetryableExceptions, retryBackoffStrategy
+    )
   }
 
   def apply(
       apiBaseUrls: Seq[String],
       urlsRetryCount: Int = DefaultUrlsRetryCount,
       startWith: Option[Int] = None,
-      optionallyRetryableExceptions: Set[OptRetryableExceptions] = Set.empty
+      optionallyRetryableExceptions: Set[OptRetryableExceptions] = Set.empty,
+      retryBackoffStrategy: Int => Unit = quadraticRandomizedRetryBackoffStrategy
    ): CrossHostApiCaller = {
-    createInstance(apiBaseUrls, urlsRetryCount, startWith, optionallyRetryableExceptions)
+    createInstance(apiBaseUrls, urlsRetryCount, startWith, optionallyRetryableExceptions, retryBackoffStrategy)
+  }
+
+  private def quadraticRandomizedRetryBackoffStrategy(retryNumber: Int): Unit = {
+    val inSeconds: Double = retryNumber * retryNumber + Random.nextDouble()
```

Comment:
I like how you added this functionality into the module, but I'm not sure if this is random enough. @benedeki what do you think, based on your experiences?

Imagine that 5 parallel processes would execute the same REST API call at the same time, then the timings would always differ only by 0-1 second. Let's say that the retryNumber is 4:
1. Fail, wait 1+(0 up to 1) seconds
2. Fail, wait 4+(0 up to 1) seconds
3. Fail, wait 9+(0 up to 1) seconds
4. Fail, wait 16+(0 up to 1) seconds.

The random part will be more and more insignificant over time. Something like this is more 'random' (=randomness will be slightly more impactful): `retryNumber * retryNumber + Random.nextDouble() * retryNumber`

But perhaps your solution is good enough as well for edge case scenarios.

Reply:
I'm open to any changes to the formula, I like `retryNumber * retryNumber + Random.nextDouble() * retryNumber` as well (now I'm thinking that this one is better than the current one), we can discuss it tomorrow/wait for @benedeki response.